plugins/module_utils/network/sonic/config/ntp/ntp.py

```diff
@@ -0,0 +1,550 @@
+        else:
+            commands = []
+
+        return commands, requests
+
+    def _state_deleted(self, want, have):
+        """ The command generator when state is deleted
+
+        :param want: the objects from which the configuration should be removed
+        :param obj_in_have: the current configuration as a dictionary
+        :param interface_type: interface type
+        :rtype: A list
+        :returns: the commands necessary to remove the current configuration
+                  of the provided objects
+        """
+        diff = get_diff(want, have, TEST_KEYS_DEL)
+
+        want_none = {'enable_ntp_auth': None, 'ntp_keys': None,
+                     'servers': None, 'source_interfaces': [],
+                     'trusted_keys': None, 'vrf': None}
+        want_any = get_diff(want, want_none, TEST_KEYS_DEL)
+        # if want_any is none, then delete all NTP configurations
+
+        if not want_any:
+            commands = have
+        else:
+            if not diff:
+                commands = want_any
+            else:
+                commands = get_diff(want_any, diff, TEST_KEYS_DEL)
+
+        commands = self.preprocess_delete_commands(commands, have)
+
+        requests = []
+        if commands:
+            requests = self.get_delete_requests(commands, have)
+
+        if len(requests) > 0:
+            commands = update_states(commands, "deleted")
+        else:
+            commands = []
+
+        return commands, requests
+
+    def preprocess_want(self, want, state):
+
+        if 'source_interfaces' in want:
+            want['source_interfaces'] = normalize_interface_name_list(want['source_interfaces'], self._module)
+
+        if state == 'deleted':
+            enable_auth_want = want.get('enable_ntp_auth', None)
+            if enable_auth_want is not None:
+                want['enable_ntp_auth'] = True
+
+            if 'servers' in want and want['servers'] is not None:
+                for server in want['servers']:
+                    server.pop('key_id')
+                    server.pop('minpoll')
+                    server.pop('maxpoll')
+            if 'ntp_keys' in want and want['ntp_keys'] is not None:
+                for ntp_key in want['ntp_keys']:
+                    ntp_key.pop('encrypted')
+                    ntp_key.pop('key_type')
+                    ntp_key.pop('key_value')
+
+        elif state == 'merged':
+            if 'servers' in want and want['servers'] is not None:
+                for server in want['servers']:
+                    if not server['key_id']:
+                        server.pop('key_id')
+                    if not server['minpoll']:
+                        server.pop('minpoll')
+                    if not server['maxpoll']:
+                        server.pop('maxpoll')
+
+    def get_create_requests(self, configs, have):
+
+        requests = []
+
+        enable_auth_config = configs.get('enable_ntp_auth', None)
+        if enable_auth_config is not None:
```

Comment:
Just curious: Is there a reason for this first config type to use:

```
if enable_auth_config is not None:
```
if enable_auth_config is not None:
```

instead of:

```
if enable_auth_config:
```

(The second form is used for the following additional configuration categories.)

Reply:
This is to check if enable_auth_config is present and enable_auth_config itself may have value True or False.